### ISSUE TITLE
[osx, parallax, wx3] Crash upon right-clicking on an a fit item in ships/fittings sidebar for the second time

### ISSUE THREAD

**Issue by noenedrops** [DarkFenX/Pyfa#424]
Steps to reproduce:

1. open pyfa
2. right-click on a fitting name, a booster menu will appear
3. click outside that menu so it hides
4. right-click on any region of the fit item again

On the second right-click the app just closes.

**Comment by thorr18** [comment]
Duplicated this issue with 1.16.x -mac-wx3 , and verified it does not occur on 1.15.x

**Comment by blitzmann** [comment]
Oh fun times:

```
2015-11-09 20:35:50.761 Python[692:12643] *** Assertion failure in -[wxNSMenuItem setSubmenu:], /SourceCache/AppKit/AppKit-1348.17/Menus.subproj/NSMenuItem.m:886
2015-11-09 20:35:50.761 Python[692:12643] Menu to be set as submenu is already a submenu of some menu.
2015-11-09 20:35:50.773 Python[692:12643] (
	0   CoreFoundation                      0x92cd8c63 __raiseError + 195
	1   libobjc.A.dylib                     0x96d53a2a objc_exception_throw + 276
	2   CoreFoundation                      0x92cd8aea +[NSException raise:format:arguments:] + 138
	3   Foundation                          0x90bfc92f -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 118
	4   AppKit                              0x98580e7a -[NSMenuItem setSubmenu:] + 222
	5   libwx_osx_cocoau-3.0.0.2.0.dylib    0x00f1f7e8 _ZN14wxMenuItemImpl6CreateEP10wxMenuItemP6wxMenuiRK8wxStringP18wxAcceleratorEntryS6_10wxItemKindS3_ + 216
	6   libwx_osx_cocoau-3.0.0.2.0.dylib    0x00e7ba6a _ZN10wxMenuItemC2EP6wxMenuiRK8wxStringS4_10wxItemKindS1_ + 362
	7   libwx_osx_cocoau-3.0.0.2.0.dylib    0x00e7c02c _ZN14wxMenuItemBase3NewEP6wxMenuiRK8wxStringS4_10wxItemKindS1_ + 76
	8   _core_.so                           0x00ab6f9d _wrap_Menu_AppendSubMenu + 397
	9   Python                              0x00133902 PyCFunction_Call + 358
	10  Python                              0x00180d54 PyEval_EvalFrameEx + 12150
	11  Python                              0x0017db9c PyEval_EvalCodeEx + 1211
	12  Python                              0x00183d50 _PyEval_SliceIndex + 752
	13  Python                              0x00180d78 PyEval_EvalFrameEx + 12186
	14  Python                              0x0017db9c PyEval_EvalCodeEx + 1211
	15  Python                              0x00122ef6 PyFunction_SetClosure + 832
	16  Python                              0x00104116 PyObject_Call + 99
	17  Python                              0x0010ebd3 PyMethod_New + 1177
	18  Python                              0x00104116 PyObject_Call + 99
	19  Python                              0x00183522 PyEval_CallObjectWithKeywords + 97
	20  _core_.so                           0x00a068ae _ZN12wxPyCallback12EventThunkerER7wxEvent + 334
	21  libwx_osx_cocoau-3.0.0.2.0.dylib    0x00cb3def _ZNK16wxAppConsoleBase16CallEventHandlerEP12wxEvtHandlerR14wxEventFunctorR7wxEvent + 95
	22  libwx_osx_cocoau-3.0.0.2.0.dylib    0x00e05f08 _ZN12wxEvtHandler23ProcessEventIfMatchesIdERK21wxEventTableEntryBasePS_R7wxEvent + 104
	23  libwx_osx_cocoau-3.0.0.2.0.dylib    0x00e072e8 _ZN12wxEvtHandler23SearchDynamicEventTableER7wxEvent + 88
	24  libwx_osx_cocoau-3.0.0.2.0.dylib    0x00e08514 _ZN12wxEvtHandler11TryHereOnlyER7wxEvent + 52
	25  libwx_osx_cocoau-3.0.0.2.0.dylib    0x00e085bc _ZN12wxEvtHandler19ProcessEventLocallyER7wxEvent + 60
	26  libwx_osx_cocoau-3.0.0.2.0.dylib    0x00e08631 _ZN12wxEvtHandler12ProcessEventER7wxEvent + 81
	27  libwx_osx_cocoau-3.0.0.2.0.dylib    0x00e07f17 _ZN12wxEvtHandler18SafelyProcessEventER7wxEvent + 39
	28  libwx_osx_cocoau-3.0.0.2.0.dylib    0x01072feb _ZNK12wxWindowBase17HandleWindowEventER7wxEvent + 27
	29  libwx_osx_cocoau-3.0.0.2.0.dylib    0x00f40217 _ZN17wxWidgetCocoaImpl18DoHandleMouseEventEP7NSEvent + 87
	30  libwx_osx_cocoau-3.0.0.2.0.dylib    0x00f3ffe2 _ZN17wxWidgetCocoaImpl10mouseEventEP7NSEventP6NSViewPv + 66
	31  libwx_osx_cocoau-3.0.0.2.0.dylib    0x00f3ac55 _Z16wxOSX_mouseEventP6NSViewP13objc_selectorP7NSEvent + 53
	32  AppKit                              0x98dfa228 -[NSWindow _reallySendEvent:isDelayedEvent:] + 939
	33  AppKit                              0x986df3ca -[NSWindow sendEvent:] + 526
	34  libwx_osx_cocoau-3.0.0.2.0.dylib    0x00f26e8d -[wxNSWindow sendEvent:] + 157
	35  AppKit                              0x986db9c4 -[NSApplication sendEvent:] + 4765
	36  libwx_osx_cocoau-3.0.0.2.0.dylib    0x00e1bbaa -[wxNSApplication sendEvent:] + 90
	37  AppKit                              0x985fbbfc -[NSApplication run] + 1003
	38  libwx_osx_cocoau-3.0.0.2.0.dylib    0x00f15ed5 _ZN14wxGUIEventLoop8OSXDoRunEv + 277
	39  libwx_osx_cocoau-3.0.0.2.0.dylib    0x00dcc4a6 _ZN13wxCFEventLoop5DoRunEv + 38
	40  libwx_osx_cocoau-3.0.0.2.0.dylib    0x00cf534a _ZN15wxEventLoopBase3RunEv + 74
	41  libwx_osx_cocoau-3.0.0.2.0.dylib    0x00cb5b8f _ZN16wxAppConsoleBase8MainLoopEv + 95
	42  _core_.so                           0x009fe2a3 _ZN7wxPyApp8MainLoopEv + 83
	43  _core_.so                           0x00a4a251 _wrap_PyApp_MainLoop + 97
	44  Python                              0x00133902 PyCFunction_Call + 358
	45  Python                              0x00180d54 PyEval_EvalFrameEx + 12150
	46  Python                              0x0017db9c PyEval_EvalCodeEx + 1211
	47  Python                              0x00122ef6 PyFunction_SetClosure + 832
	48  Python                              0x00104116 PyObject_Call + 99
	49  Python                              0x0010ebd3 PyMethod_New + 1177
	50  Python                              0x00104116 PyObject_Call + 99
	51  Python                              0x0018087f PyEval_EvalFrameEx + 10913
	52  Python                              0x00183def _PyEval_SliceIndex + 911
	53  Python                              0x00180d78 PyEval_EvalFrameEx + 12186
	54  Python                              0x0017db9c PyEval_EvalCodeEx + 1211
	55  Python                              0x0017d6dc PyEval_EvalCode + 87
	56  Python                              0x0019b757 PyParser_ASTFromFile + 346
	57  Python                              0x0019b7f9 PyRun_FileExFlags + 138
	58  Python                              0x0019b38e PyRun_SimpleFileExFlags + 773
	59  Python                              0x0019af2c PyRun_AnyFileExFlags + 108
	60  Python                              0x001ad92f Py_Main + 3427
	61  libdyld.dylib                       0x946cd6d9 start + 1
)
```

It looks like we try to attach the menu twice. Only happens when the "Set Booster" menu is available (ie: when we have a fit loaded).

**Comment by blitzmann** [comment]
fixed 4d701f561e8f14db71fa15450c3
